flocker/acceptance/testtools.py

```diff
@@ -789,6 +839,15 @@ def decorator(test_method):
         """
         def call_test_method_with_cluster(cluster, test_case, args, kwargs):
             kwargs['cluster'] = cluster
+            if required_backend:
+                backend_type = get_dataset_backend(test_case)
+                if backend_type != required_backend:
+                    raise SkipTest(
+                        'This test requires backend type {} but is being run '
+                        'on {}.'.format(backend_type.name,
```

Comment:
These are backwards: http://build.clusterhq.com/builders/flocker%2Facceptance%2Frackspace%2Fcentos-7%2Frackspace/builds/3001/steps/trial/logs/stdio

I'll fix that when I address other review comments.